base.mk

```diff
@@ -116,7 +116,7 @@ BUILD ?= release
 CFLAGS += -ffunction-sections -fdata-sections
 LDFLAGS += -Xlinker --gc-sections
 ifeq ($(BUILD), debug)
-CFLAGS += -O0 -g -DDEBUG
+CFLAGS += -Og -g -DDEBUG
```

Comment:
Yes, the problem is that for the level of debugging we need in the majority of the cases, having any optimizations in place for debug builds mislead us into issues. Let's keep the default debug build as -O0 as this is something easily modifiable when one needs to change it into -Og, ok?

We could consider having a third BUILD= option that is optimized-debug or similar, but I don't think it pays off.